src/templates/components/Fabric.svelte

```diff
@@ -26,33 +34,74 @@
 	export let MobileLayer2BackgroundPosition: string;
 	export let MobileLayer3BackgroundImage: string;
 	export let MobileLayer3BackgroundPosition: string;
+	export let VideoURL: GAMVariable;
+	export let VideoBackupImage: GAMVariable;
+	export let MobileVideoBackupImage: GAMVariable;
+	export let VideoURLMobile: GAMVariable;
+	export let VideoAlignment: GAMVariable;
 	export let isXL = false;
 
-	const isMobile = window.matchMedia('(max-width: 739px)').matches;
+	const isMobile = window.innerWidth < 740;
 	const isTablet = window.matchMedia(
 		'(min-width: 740px) and (max-width: 979px)',
 	).matches;
 
-	const [backgroundImage, backgroundPosition, backgroundRepeat] = isMobile
-		? [
-				MobileBackgroundImage,
-				MobileBackgroundImagePosition,
-				MobileBackgroundImageRepeat,
-			]
-		: [BackgroundImage, BackgroundImagePosition, BackgroundImageRepeat];
-
-	const checkScrollType = (scrollType: string): string =>
-		scrollType === 'parallax' && (isMobile || isTablet) ? 'fixed' : scrollType;
-
-	post({
-		type: 'background',
-		value: {
-			scrollType: checkScrollType(BackgroundScrollType),
-			backgroundColor: BackgroundColour,
-			backgroundImage: `url('${backgroundImage}')`,
-			backgroundRepeat,
-			backgroundPosition,
-		},
+	const video: Writable<HTMLVideoElement | undefined> = writable();
+	let played = false;
+
+	const posterImage = isMobile ? MobileVideoBackupImage : VideoBackupImage;
+	const videoSrc = isMobile ? VideoURLMobile : VideoURL;
+
+	onMount(() => {
```

Comment:
I don't think the onMount is needed?

Reply:
@Jakeii I think in this case  the onMount function is used to handle video loading, since this Fabric components scope and capabilities have been widened to include the video functionality,  the onMount function is used to conditionally execute code based on the value of showVideo. But there may be a better solution im not aware of?

Reply:
The `video.subscribe` should take care of it no need for onMount

Reply:
@Jakeii I have refacored still taking into account size xl